drv/transceivers-server/src/main.rs

```diff
@@ -408,14 +412,16 @@ impl ServerImpl {
                 None => continue,
             };
 
-            // *Always* post the thermal model over to the thermal task, so that
-            // the thermal task still has it in case of restart.  This will
-            // return a `NotInAutoMode` error if the thermal loop is in manual
-            // mode; this is harmless and will be ignored (instead of cluttering
-            // up the logs).
-            match self.thermal_api.update_dynamic_input(i, m.model) {
-                Ok(()) | Err(ThermalError::NotInAutoMode) => (),
-                Err(e) => ringbuf_entry!(Trace::ThermalError(i, e)),
+            if let Some(thermal_api) = &self.thermal_api {
+                // *Always* post the thermal model over to the thermal task, so
+                // that the thermal task still has it in case of restart.  This
+                // will return a `NotInAutoMode` error if the thermal loop is in
+                // manual mode; this is harmless and will be ignored (instead of
+                // cluttering up the logs).
+                match thermal_api.update_dynamic_input(i, m.model) {
```

Comment:
similarly, this doesn't really need to be a runtime check; if we followed my suggestion from above, this could instead be:
```suggestion
            #[cfg(feature = "thermal-control")]
            {
                // *Always* post the thermal model over to the thermal task, so
                // that the thermal task still has it in case of restart.  This
                // will return a `NotInAutoMode` error if the thermal loop is in
                // manual mode; this is harmless and will be ignored (instead of
                // cluttering up the logs).
                match self.thermal_api.update_dynamic_input(i, m.model) {
```